upload-api/functions/validate-email.jsx

```diff
@@ -215,21 +216,32 @@ export async function validateEmailPost(request) {
   }
 
   const { email, audience, ucan } = authorizeResult.ok
-  console.log("CHECKING PLAN")
 
   const planCheckResult = await context.customerStore.get({
     customer: DidMailto.fromEmail(email),
   })
+  let isReferred = false
+  try {
+    // if we can find a referral code for this user, offer them a free trial
+    if ((await context.referralStore.getReferredBy(email)).refcode) {
+      isReferred = true
+    }
+  } catch (e){
+    // if we fail here, log the error and move on
+    console.warn('encountered an error checking the referrals service, please see the error logs for more information')
+    console.error(e)
+  }
   let stripePricingTableId
   let stripePublishableKey
-  console.log("CHECKED PLAN", JSON.stringify(planCheckResult))
 
   if (!planCheckResult.ok?.product) {
-    stripePricingTableId = context.stripePricingTableId
     stripePublishableKey = context.stripePublishableKey
+    if (isReferred) {
+      stripePricingTableId = context.stripeFreeTrialPricingTableId
+    } else {
+      stripePricingTableId = context.stripePricingTableId
+    }
```

Comment:
**suggestion:** …I'm not totally sure what generates these errors for us, but whatever it is, I agree with it. 😄 Let's prefer the ternary to mutating a variable.